hkdfchacha20poly1305/hkdfchachapoly.go

```diff
@@ -0,0 +1,87 @@
+package hkdfchacha20poly1305
```

Comment:
Can we a have a high level overview here? It will create a neat godoc entry and helps the user understand what is going on below. Just a few sentences